cmd/deploy.go

```diff
@@ -0,0 +1,144 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package cmd defines command line utilities for ghpc
+package cmd
+
+import (
+	"fmt"
+	"hpc-toolkit/pkg/config"
+	"hpc-toolkit/pkg/shell"
+	"log"
+	"path/filepath"
+
+	"github.com/spf13/cobra"
+)
+
+func init() {
+	artifactsFlag := "artifacts"
+
+	deployCmd.Flags().StringVarP(&artifactsDir, artifactsFlag, "a", "", "Artifacts output directory (automatically configured if unset)")
+	deployCmd.MarkFlagDirname(artifactsFlag)
+
+	autoApproveFlag := "auto-approve"
+	deployCmd.Flags().BoolVarP(&autoApprove, autoApproveFlag, "", false, "Automatically approve proposed changes")
```

Comment:
Resolved offline to avoid introducing a short option. Reasoning:

1. Explicitness / alignment with terraform.
2. Easier to add in future than to take away.